Mathlib/Data/Finset/Card.lean

```diff
@@ -590,6 +590,25 @@ theorem exists_ne_of_one_lt_card (hs : 1 < s.card) (a : α) : ∃ b, b ∈ s ∧
   · exact ⟨y, hy, ha⟩
 #align finset.exists_ne_of_one_lt_card Finset.exists_ne_of_one_lt_card
 
+/-- The product of the cardinalities of two Finsets is greater than 1 iff
+  one of the cardinalities is greater than 1 and both Finsets are nonempty. -/
+lemma one_lt_card_mul_card_iff {t : Finset β} : 1 < s.card * t.card ↔
+    s.Nonempty ∧ t.Nonempty ∧ (1 < s.card ∨ 1 < t.card) := by
```

Comment:
How about:
```suggestion
lemma nontrivial_prod_iff {t : Finset β} : Nontrivial (s ×ˢ t) ↔
    s.Nonempty ∧ t.Nonempty ∧ (Nontrivial s ∨ Nontrivial t) := by
```

Reply:
I've opened #7131 which adds the missing `Finset.one_lt_card_iff_nontrivial`.

I suggest we wait until its fate is decided and then reconsider the statement here.

Reply:
#7131 looks good to me, but since it doesn't change [Finset.one_lt_card](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Finset/Card.html#Finset.one_lt_card), [Finset.one_lt_card_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Finset/Card.html#Finset.one_lt_card_iff), and [Finset.exists_ne_of_one_lt_card](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Finset/Card.html#Finset.exists_ne_of_one_lt_card) to use `Nontrivial`, it shouldn't affect the choice whether to use `Nontrivial` here.

Reply:
This is obviously subjective so if you remain unconvinced I can let it go. Just before I do, I'd like to state my case more explicitly.

I claim the statement:
```lean
    Nontrivial (s ×ˢ t) ↔ s.Nonempty ∧ t.Nonempty ∧ (Nontrivial s ∨ Nontrivial t)
```
is preferable to the statement:
```lean
    1 < s.card * t.card ↔ s.Nonempty ∧ t.Nonempty ∧ (1 < s.card ∨ 1 < t.card)
```
This is partly because the statement using `Nontrivial` is how I actually think about this result; I agree this is debatable.

However now that I #7131 has made `Finset.one_lt_card_iff_nontrivial_coe` available, I do not believe there is much strength to an argument about other lemmas mentioning `1 < s.card` rather than `Nontrivial s`. Indeed the proof of my suggested statement could just be:
```lean
  simp_rw [← card_pos, ← one_lt_card_iff_nontrivial_coe, card_product]; apply Nat.one_lt_mul_iff
```

Reply:
Okay I adopted your statement but I'm not sure I'll use this lemma anymore ...